extensions/ql-vscode/.storybook/preview.js

```diff
@@ -0,0 +1,38 @@
+import { themes } from '@storybook/theming';
+import { action } from '@storybook/addon-actions';
+
+// Allow all stories/components to use Codicons
+import '@vscode/codicons/dist/codicon.css';
+
+import '../src/stories/vscode-theme.css';
```

Comment:
Instead of having our own file that we have to manage/keep up-to date etc., is there a way to get everything from VSCode automatically?

Reply:
We might be able to generate it from the VSCode source code, but I don't think we can import it from VSCode itself. 

The colors are defined [here](https://github.com/microsoft/vscode/blob/main/src/vs/platform/theme/common/colorRegistry.ts) and the styles are created [here](https://github.com/microsoft/vscode/blob/a6272c941610ec1da55f48ea425cdb4bf4868543/src/vs/workbench/contrib/webview/browser/themeing.ts#L51-L82). They are then injected into the WebView using [this](https://github.com/microsoft/vscode/blob/a6272c941610ec1da55f48ea425cdb4bf4868543/src/vs/workbench/contrib/webview/browser/webviewElement.ts#L740-L750) on the host side and [this](https://github.com/microsoft/vscode/blob/a6272c941610ec1da55f48ea425cdb4bf4868543/src/vs/workbench/contrib/webview/browser/pre/index.html#L497-L515) on the WebView side.

I think it will be a lot more work to generate the CSS variables from this than to copy the variables from VSCode once in a while.

Reply:
Ah that's a shame. I suspected it would be a can of warms, but I thought I'd ask anyway 😂 

In that case I think we should have some docs to tell people about this (maybe in the storybook part in the README?) and how to update them.

Reply:
Perhaps I can also add it to an Overview page in Storybook itself and refer to that from the CONTRIBUTING documentation?

Reply:
SGTM!

Reply:
I just had a brainwave! Feel free to ignore this if you want (totally up to you), but the actual toolkit has a storybook, so perhaps this is something they've solved already? See https://github.com/microsoft/vscode-webview-ui-toolkit/tree/main/.storybook

Reply:
I did see whether it was possible to replicate the way they are using the VSCode variables, but they are using a completely custom system. They have a system of "design tokens" (https://github.com/microsoft/vscode-webview-ui-toolkit/blob/6fc339ce26e7da15b9ba97c063f56d8a4eae85ad/src/design-tokens.ts) which specify a variable and a default. These are then used in [the styles](https://github.com/microsoft/vscode-webview-ui-toolkit/blob/6fc339ce26e7da15b9ba97c063f56d8a4eae85ad/src/button/button.styles.ts) and will automatically use either the variable or the default. I don't think that is going to work for us since we'd need to modify the way we write the CSS files and set defaults for all variables we use. It could also result in having different defaults than the toolkit, so it might also get out of sync at some point.